### PR TITLE
local-interchain: Rust Driver

### LINKED ISSUE #762: Add a Rust driver
Following #761, add a rust based client following typical rust conventions.

Feature set should match of exceed the python client

### PR DESCRIPTION
**Pull request by Reecepbcups**
Closes #762 

## summary
Adds a rust driver to local-interchain & ci testing (for those who do not want to write Go)

## reference - outside repo example
- [rust test code](https://github.com/Reecepbcups/interchaintest-rust-example/blob/main/interchaintest/src/main.rs)
- [action e2e ci](https://github.com/Reecepbcups/interchaintest-rust-example/blob/main/.github/workflows/e2e.yml)

## considerations
- We could post localic_std to crates.io, but using it in the git repo works too (see reference above)
- the actions ci requires a bunch of overhead, prebuilt binaries would be nice (requires release cycles)

### TODO:
- [x] Match python spec
- [x] No Panics / unwraps, Results only
- [x] Clippy lint + fmt CI, no panics
- [x] Add Rust CI test (Match Python)
- [x] test in an external repo, ensure the workflow is simple to setup
- [x] integration docs + example
- [x] Ensure chain actions are in POST request only
- [x] properly sort types (ex: put CosmWasm in a modules mod)

## todo after merge
- The python client needs to use `Upload-Type: cosmwasm` header for store_code uploads. (per 3aab50abb64b9255814a4ce200d401a0792a50f2)
- Remove server.json dep from python client (use cli / defaults)
- Remove relayer.json need? (make optional, have relayer default, add cli arg for --relayer on start)

### CODE REVIEW

**Review by misko9 — COMMENTED** [review]
Super cool, need to play around with this and get it in some flows.

File: local-interchain/cmd/local-ic/start_chain.go [hunk 1]
```diff
@@ -27,6 +32,17 @@ var startCmd = &cobra.Command{
...
+
+		interchain.StartChain(parentDir, configPath, &interchain.AppOverrides{
+			AddressOverride: apiAddrOverride,
+			PortOverride:    apiPortOverride,
+		})
 	},
 }
+
+func init() {
+	startCmd.Flags().String(FlagAPIAddressOverride, "", "override the default API address")
```
[the remark this answers is not in the document]
Comment (Reecepbcups) [reply local-interchain/cmd/local-ic/start_chain.go]: Yes, resolved in 87d4dbaa4dbcfeef36e09ee4d9b3b4481a81c146

File: local-interchain/interchain/handlers/uploader.go [hunk 2]
```diff
@@ -18,8 +19,10 @@ type upload struct {
 
 type Uploader struct {
 	ChainId  string `json:"chain_id"`
-	KeyName  string `json:"key_name"`
-	FileName string `json:"file_name"`
+	FilePath string `json:"file_path"`
+
+	// Upload-Type: cosmwasm only
+	KeyName string `json:"key_name"`
```
[the remark this answers is not in the document]
Comment (Reecepbcups) [reply local-interchain/interchain/handlers/uploader.go]: resolved in 4211fd082d635d62a02d2314a344a93b9c2f843f

File: local-interchain/rust/localic-std/src/relayer.rs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    /// Returns `Err` if the relayer is not able to flush packets between 2 contracts.
+    pub fn flush(&self, path: &str, channel: &str) -> Result<Value, LocalError> {
+        let cmd = format!("rly transact flush {path} {channel}");
+        self.execute(cmd.as_str(), false)
+    }
+
+    /// # Errors
+    ///
+    /// Returns `Err` if the relayer is not able to create the connection between 2 contracts.
+    pub fn create_wasm_connection(
```
[the remark this answers is not in the document]
Comment (Reecepbcups) [reply local-interchain/rust/localic-std/src/relayer.rs]: Ahh yes, this was the approach I was going for and completely overlooked it in future development. Nice find

Resolved in 65222f457415b1d29a9171f9322c31a9a03f21ea (relayer create_connection is generalized, while create_wasm_connection in CosmWasm impl handles the wasm. prefix)

[the remark this answers is not in the document]
Comment (Reecepbcups) [reply local-interchain/rust/localic-std/src/relayer.rs]: and 5a9b43795b213a3c16acfb72550dc72a17f9e2a7 to avoid confusion as you mentioned

**Review by misko9 — APPROVED** [review]
Looking good!

**Pull request merged**